earth_enterprise/src/fusion/autoingest/Asset.h

```diff
@@ -93,6 +94,7 @@ template <>
 inline void
 Asset::DoBind(const std::string &ref, bool checkFileExistenceFirst) const
 {
+  khTargetedLock lock(ref);
```

Comment:
This was added in order to prevent two threads from hitting this section simultaneously for the same file and both independently loading the file into memory.  Is this overkill to prevent something that *shouldn't* cause any problems?